crates/nu-cmd-extra/src/extra/bits/mod.rs

```diff
@@ -95,7 +100,68 @@ fn get_input_num_type(val: i64, signed: bool, number_size: NumberBytes) -> Input
                     InputNumType::Eight
                 }
             }
-            NumberBytes::Invalid => InputNumType::Four,
         }
     }
 }
+
+fn binary_op<F>(lhs: &Value, rhs: &Value, little_endian: bool, f: F) -> Value
+where
+    F: Fn((i64, i64)) -> i64,
+{
+    let span = lhs.span();
+    match (lhs, rhs) {
+        (Value::Int { val: lhs, .. }, Value::Int { val: rhs, .. }) => {
+            Value::int(f((*lhs, *rhs)), span)
+        }
+        (Value::Binary { val: lhs, .. }, Value::Binary { val: rhs, .. }) => {
+            let (lhs, rhs, max_len, min_len) = match (lhs.len(), rhs.len()) {
+                (max, min) if max > min => (lhs, rhs, max, min),
+                (min, max) => (rhs, lhs, min, max),
+            };
+
+            let pad = iter::repeat(0).take(max_len - min_len);
+
+            let mut a;
+            let mut b;
+
+            let padded: &mut dyn Iterator<Item = u8> = if little_endian {
+                a = pad.chain(rhs.iter().copied());
+                &mut a
+            } else {
+                b = rhs.iter().copied().chain(pad);
+                &mut b
+            };
+
+            let bytes: Vec<u8> = lhs
+                .iter()
```

Comment:
Does this handle the case when `lhs` is smaller than `rhs` and the user wants to use little endian? It feels like `lhs` should be padded too